coreax/kernel.py

```diff
@@ -348,6 +348,45 @@ def calculate_kernel_matrix_row_sum_mean(
         return self.calculate_kernel_matrix_row_sum(x, max_size) / (1.0 * x.shape[0])
 
 
+class LinearKernel(Kernel):
+    """Define a linear kernel."""
+
+    length_scale: float = 1.0
+    output_scale: float = 1.0
+
+    @override
+    def compute_elementwise(
+        self,
+        x: ArrayLike,
+        y: ArrayLike,
+    ) -> Array:
+        return x.dot(y)
+
+    @override
+    def grad_x_elementwise(
+        self,
+        x: ArrayLike,
+        y: ArrayLike,
+    ) -> Array:
+        return y
```

Comment:
Similar to above. To actually be returning an `Array`, y needs to be cast using `jnp.asarray(y)`. Same for grad_y_elementwise.